core/win32/inject.c

```diff
@@ -1082,11 +1082,12 @@ generate_switch_mode_jmp_to_hook(HANDLE phandle, byte *local_code_buf,
     instr_t *restore_esp =
         INSTR_CREATE_mov_ld(GDC, opnd_create_reg(REG_ESP),
                             OPND_CREATE_MEM32(REG_NULL, (int)(size_t)mode_switch_data));
-	byte* eaxpos =(byte*) ((size_t)mode_switch_data) + 8;
-	// Restoring eax back, which is storing routine address that needs to be passed to RtlUserStartThread
-	instr_t *restore_eax = INSTR_CREATE_mov_ld(GDC, opnd_create_reg(REG_EAX), 
-		OPND_CREATE_MEM32(REG_NULL, (int)(size_t)eaxpos));
-	
+    const byte *eax_saved_offset = (byte *)((size_t)mode_switch_data) + 4;
+    // Restoring eax back, which is storing routine address that needs to be passed to
+    // RtlUserStartThread
```

Comment:
From other comment thread: would be good to explain that we do not know why eax is clobbered in a `XXX` comment since this may indicate some bug or problem somewhere else.

Reply:
Done